src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -466,11 +467,278 @@ type Moon {
+                            },
+                            {
+                              ""action"": ""delete""
+                            }
+                          ]
+                        }
+                      ],
+                        ""mappings"": {
+                          """ + entityBackingColumn + @""": """ + entityExposedName + @"""
+                        }
+                      }
+                  }
+              }");
+
+        }
+
+            [TestCleanup]
         public void CleanupAfterEachTest()
         {
             TestHelper.UnsetAllDABEnvironmentVariables();
+            if (File.Exists(HOT_RELOAD_CONFIG_FILE_NAME))
+            {
+                File.Delete(HOT_RELOAD_CONFIG_FILE_NAME);
+            }
+        }
+
+        /// <summary>
+        /// Start DAB with the pre-formed config file, assert that rest and GQL
+        /// requests for the book database object returns the expected results.
+        /// Then hot-reload the config runtime paths both for rest and GQL, and
+        /// again assert that the requests return the expected results against
+        /// the book database object.
+        /// </summary>
+        /// <returns></returns>
+        [TestCategory(MSSQL_ENVIRONMENT)]
+        [TestMethod("Hot-reload runtime paths.")]
+        public async Task HotReloadConfigRuntimePathsEndToEndTest()
+        {
+            // Arrange
+            string restBookContents = $"{{\"value\":{BOOK_DBO_CONTENTS}}}";
+            string restPath = "/restApi";
+            string gQLPath = "/gQLApi";
+
+            string graphQLQueryName = "books";
+            string graphQLQuery = @"{
+                books(first: 100) {
+                    items {
+                        id
+                        title
+                        publisher_id
+                    }
+                }
+            }";
+
+            GenerateConfigFile(connectionString: $"{GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL)}");
+
+            TestServer server = new(Program.CreateWebHostBuilder(new string[] { "--ConfigFileName", HOT_RELOAD_CONFIG_FILE_NAME }));    
+            HttpClient httpClient = server.CreateClient();
+            RuntimeConfigProvider configProvider = server.Services.GetService<RuntimeConfigProvider>();
+
+            // Act
+            HttpResponseMessage result = await httpClient.GetAsync("/rest/Book");
+            string initialRestContents = await result.Content.ReadAsStringAsync();
+            JsonElement initialGQLContents = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                httpClient,
+                configProvider,
+                graphQLQueryName,
+                graphQLQuery);
+
+            // Assert
+            Assert.IsTrue(SqlTestHelper.JsonStringsDeepEqual($"{restBookContents}", initialRestContents));
+            SqlTestHelper.PerformTestEqualJsonStrings(BOOK_DBO_CONTENTS, initialGQLContents.GetProperty("items").ToString());
```

Comment:
why do we need to validate that all the book contents are returned for a test that's just checking the REST path? Would check HTTP200 response be sufficient here?

Reply:
ya that should be fine.